tests/test_extension_robotstxtparser.py

```diff
@@ -0,0 +1,143 @@
+from twisted.trial import unittest
+from scrapy.utils.python import to_native_str
+from scrapy.extensions.robotstxtparser import PythonRobotParser, ReppyRobotParser, RerpRobotParser
+
+class PythonRobotParserTest(unittest.TestCase):
+    def test_allowed(self):
+        robotstxt_content = ("User-agent: * \n"
+                            "Disallow: /disallowed \n"
+                            "Allow: /allowed \n"
+                            "Crawl-delay: 10".encode('utf-8'))
+        rp = PythonRobotParser(robotstxt_content)
+        self.assertTrue(rp.allowed("https://www.example.com/allowed", "*"))
+        self.assertFalse(rp.allowed("https://www.example.com/disallowed", "*"))
+
+    def test_sitemaps(self):
+        robotstxt_content = ("User-agent: * \n"
+                            "Disallow: /disallowed \n"
+                            "Allow: /allowed \n"
+                            "Crawl-delay: 10".encode('utf-8'))
+        rp = PythonRobotParser(robotstxt_content)
+        self.assertTrue(not list(rp.sitemaps()))
+
+    def test_preferred_host(self):
+        robotstxt_content = ("User-agent: * \n"
+                            "Disallow: /disallowed \n"
+                            "Allow: /allowed \n"
+                            "Crawl-delay: 10".encode('utf-8'))
+        rp = PythonRobotParser(robotstxt_content)
+        self.assertTrue(rp.preferred_host() is None)
+
+    def test_crawl_delay(self):
+        robotstxt_content = ("User-agent: * \n"
+                            "Disallow: /disallowed \n"
+                            "Allow: /allowed \n"
+                            "Crawl-delay: 10".encode('utf-8'))
+        rp = PythonRobotParser(robotstxt_content)
+        from six.moves.urllib_robotparser import RobotFileParser
+        if hasattr(RobotFileParser, "crawl_delay"):
+            self.assertTrue(rp.crawl_delay("*") == 10.0)
+        else:
+            self.assertTrue(rp.crawl_delay("*") is None)
+
+        robotstxt_content = ("User-agent: * \n"
```

Comment:
I think, these should be two separate test cases `test_crawl_delay` and `test_no_crawl_delay`